src/sqlitetablemodel.cpp

```diff
@@ -432,6 +432,26 @@ bool SqliteTableModel::removeRows(int row, int count, const QModelIndex& parent)
     return true;
 }
 
+QModelIndex SqliteTableModel::dittoRecord(int old_row)
+{
+    int firstEditedColumn = 0;
+    int new_row = rowCount() - 1;
+
+    sqlb::Table t = sqlb::Table::parseSQL(m_db->getObjectByName(m_sTable).getsql()).first;
+    for (int col = 0; col < t.fields().size(); ++col) {
+        if (!t.fields().at(col)->primaryKey()) {
+            if (!firstEditedColumn)
+                firstEditedColumn = col + 1;
+
+            QVariant value = m_data.at(old_row).value(col + 1);
+            if (!value.isNull())
```

Comment:
I'm not sure about this line. What's your intention here? My worries are these:
Create a table like this one
```CREATE TABLE test(a INT, b INT DEFAULT 5);```
and fill it with some data
```INSERT INTO test VALUES(1, NULL);```
Now, duplicate that row and the two rows won't be identical.